snooty/util.py

```diff
@@ -192,6 +192,16 @@ def __len__(self) -> int:
         return sum(len(w) for w in self.directories.values())
 
 
+def option_string(argument: Optional[str]) -> Optional[str]:
+    """
+    Check for a valid string option and return it. If no argument is given,
+    return None.
+    """
+    if argument and argument.strip():
+        return argument
+    return None
```

Comment:
Let's raise a diagnostic before returning `None` to warn writers that they're using the option incorrectly!

Reply:
Specifically, I think your only option is to raise ValueError

Reply:
I've added a small check to parser.py instead, so that we don't have to raise a ValueError (which might break current builds). Instead, I added a diagnostic for start-after and end-before.

Reply:
OK, so I did end up adding the `raise ValueError`, per Andrew's suggestion. Corresponding tests are in `test_util.py`.